### PR TITLE
ISPN-10002 Image configurable via CR spec

### PR DESCRIPTION
**Pull request by rigazilla**
This PR makes configurable the Image name for pods containers

### CODE REVIEW

File: pkg/controller/infinispan/infinispan_controller.go [hunk 1]
```diff
@@ -165,6 +165,16 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 	}
 
+	// Check if pods container runs the right image
```
Comment (galderz) [inline pkg/controller/infinispan/infinispan_controller.go]: What's this check about?

**Comment by galderz** [comment]
@rigazilla I've seen both PRs fail with timeout waiting for first pods to appear. I've seen that happening locally when there are leftover things. Dunno why might be happening here. I'll try to test locally and see if I can add more logging when timeout happens.

**Comment by galderz** [comment]
Needs rebasing

File: pkg/controller/infinispan/infinispan_controller.go (hunk 1 above)
Comment (rigazilla) [reply pkg/controller/infinispan/infinispan_controller.go]: Suppose that we have a 9.4.0 running cluster and the user changes the image to 9.4.5 and deploys cr. The Reconcile policy needs to handle this in some way, but we need some use cases definition first: i.e. is this a cluster upgrade use case?

Basically I added this check as a todo flag atm.

File: pkg/apis/infinispan/v1/infinispan_types.go [hunk 2]
```diff
@@ -6,7 +6,9 @@ import (
 
 // InfinispanSpec defines the desired state of Infinispan
 type InfinispanSpec struct {
-	Size int32 `json:"size"`
+	Size        int32  `json:"size"`
+	ClusterName string `json:"clustername"`
```
Comment (galderz) [inline pkg/apis/infinispan/v1/infinispan_types.go]: This shouldn't be here, it was removed by ba950d6860bae1bcaa795cfee407f6ff8175e874

Comment (rigazilla) [reply pkg/apis/infinispan/v1/infinispan_types.go]: weird, fixed

**Pull request merged**